### PR TITLE
feat: Allow users to add more Audiences to OpenID Connect

### LINKED ISSUE #1145: enable_irsa=true creates OIDC provider with audience pointing to global sts endpoint (sts.amazonaws.com)
# I have issues

## I'm submitting a...

* [ ] bug report
* [x ] feature request
* [ ] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

EKE- 1.18, Fargate

when enable_irsa = true an OIDC provider is created correctly and the audience is set to "sts.amazonaws.com". However, in most corporate network one is encouraged to use local endpoint "sts.eu-central-1.amazonaws.com". In most of the corporate network I have experienced the  "sts.amazonaws.com" is blocked which creates issues for role assumption by containers.

## What's the expected behavior?

Either based on region setting of the EKS cluster or through explicit setting of sts endpoint it would be nice to have to be able to configure OIDC audience (sts endpoint) explicitly with some property like

oidc_audience = sts.xxxx.com (if irsa is set tot true)
 

## Environment details

* Affected module version:
* OS: Linux
* Terraform version: 0.13.4

### PR DESCRIPTION
**Pull request by cabrinha**
# PR o'clock

Fixes #1145

## Description

I'd like to add more audiences to the OpenID Connect Provider: https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1145

### Checklist

- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by cabrinha** [comment]
@barryib can I get a review on this one?

**Review by daroga0002 — APPROVED** [review]

File: local.tf [hunk 1]
```diff
@@ -44,7 +44,7 @@ locals {
   )
 
   ec2_principal = "ec2.${data.aws_partition.current.dns_suffix}"
-  sts_principal = "sts.${data.aws_partition.current.dns_suffix}"
+  sts_principal = concat(["sts.${data.aws_partition.current.dns_suffix}"], var.openid_connect_audiences)
```
Comment (antonbabenko) [inline local.tf:47]: Shouldn't we remove empty elements from the list by wrapping this with `compact`? 

```suggestion
  sts_principal = compact(concat(["sts.${data.aws_partition.current.dns_suffix}"], var.openid_connect_audiences))
```

Comment (daroga0002) [reply local.tf:47]: yup, you are right

**Comment by antonbabenko** [comment]
Hi @cabrinha and thanks for the PR!

[v17.5.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/v17.5.0) has been just released.

**Comment by lisfo4ka** [comment]
@antonbabenko @daroga0002 folks, maybe it's too late to claim, but I assume that the name of the variable with the client id list should be something other than `sts_principal` since it can contain not only STS endpoint but any client ID that identifies the application. See an example, in the doc https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_openid_connect_provider
So I'm back to propose the change from my PR to have the separate variable for a full list with client ids. What do you think about this?

**Comment by daroga0002** [comment]
if I understand correctly you want just to change a `locals` variable name?

I think as this PR was arleady merged and there was created release input variable should stay as is (`openid_connect_audiences`) but locals probably can be adjusted as this will make sense and will not cause any user facing changes.

**Comment by antonbabenko** [comment]
@lisfo4ka Isn't `sts` a required item in the context of this module? I mean, sometimes one may want to have another value but in this module, EKS always expects to have at least one `sts...` item, right?

**Comment by lisfo4ka** [comment]
@daroga0002 yes, you're right, I've meant just a local variable renaming.

But I see now that it's really not so important change since for the IRSA option the client id list will contain STS endpoints only. Thanks, @antonbabenko.

So let's leave as it is?

**Comment by antonbabenko** [comment]
We can rename `locals` name without any problems. `client_id_list` as a local name, and value wrapped with `distinct()` would be great. Please make a PR.

**Comment by lisfo4ka** [comment]
@antonbabenko  @daroga0002  please, find the discussed changes in #1561.
I've tested it with the latest `master`.

**Pull request merged**